enterprise/internal/authz/perforce/protects.go

```diff
@@ -366,6 +367,7 @@ func fullRepoPermsScanner(logger log.Logger, perms *authz.ExternalUserPermission
 
 	return &protectsScanner{
 		processLine: func(line p4ProtectLine) error {
+			logger.Debug("Processing parsed line", log.String("match", line.match), log.Bool("isExclusion", line.isExclusion))
```

Comment:
You can ensure the same context is included on all lines by creating a sublogger here:

```suggestion
			lineLogger := logger.With(log.String("match", line.match), log.Bool("isExclusion", line.isExclusion))
			lineLogger.Debug("Processing parsed line")
```

And using `lineLogger` throughout.

Reply:
Thanks for the tip. In this instance I don't think we want to repeat the parsed line in every log.

Reply:
On second thought, this may help me debug even if it is a bit noiser. PR incoming :)